packages/devtools_app/lib/src/flutter/extent_delegate_list.dart

```diff
@@ -0,0 +1,464 @@
+/// specified by the [extentDelegate].
+///
+/// This class is inspired by the functionality in [ListView] where
+/// `itemExtent` is specified. The difference is the extentDelegate provided
+/// here specifies different extents for each item in the list and provides
+/// the ability to animate extents without rebuilding the list. You should use
+/// ListView instead for the simpler case where all items have the same extent.
+///
+/// Using this class is more efficient than using a ListView without specifying
+/// itemExtent as only items visible on screen need to be built and laid out.
+/// This class is more robust than ListView for cases where ListView items off
+/// screen need to be animated.
+class ExtentDelegateListView extends BoxScrollView {
+  const ExtentDelegateListView({
+    Key key,
+    Axis scrollDirection = Axis.vertical,
+    bool reverse = false,
+    ScrollController controller,
+    bool primary,
+    ScrollPhysics physics,
+    bool shrinkWrap = false,
+    EdgeInsetsGeometry padding,
+    @required this.childrenDelegate,
+    @required this.extentDelegate,
+    int semanticChildCount,
+  })  : assert(childrenDelegate != null),
+        super(
+          key: key,
+          scrollDirection: scrollDirection,
+          reverse: reverse,
+          controller: controller,
+          primary: primary,
+          physics: physics,
+          shrinkWrap: shrinkWrap,
+          padding: padding,
+          semanticChildCount: semanticChildCount,
+        );
+
+  /// A delegate that provides the children for the [ExtentDelegateListView].
+  final SliverChildDelegate childrenDelegate;
+
+  /// A delegate that provides item extents for the children of the
+  ///  [ExtentDelegateListView].
+  final ExtentDelegate extentDelegate;
+
+  @override
+  Widget buildChildLayout(BuildContext context) {
+    return SliverExtentDelegateList(
+      delegate: childrenDelegate,
+      extentDelegate: extentDelegate,
+    );
+  }
+}
+
+/// A sliver that places multiple box children in a linear array.
+///
+/// The  main axis extents on each child are specified by a delegate.
+///
+/// This class is inspired by [SliverFixedExtentList] which provides similar
+/// functionality for the case where all items have the same extent.
+class SliverExtentDelegateList extends SliverMultiBoxAdaptorWidget {
+  /// Creates a sliver that places box children with the same main axis extent
+  /// in a linear array.
+  const SliverExtentDelegateList({
+    Key key,
+    @required SliverChildDelegate delegate,
+    @required this.extentDelegate,
+  }) : super(key: key, delegate: delegate);
+
+  /// The extent the children are forced to have in the main axis.
+  final ExtentDelegate extentDelegate;
+
+  @override
+  RenderSliverExtentDelegateBoxAdaptor createRenderObject(
+      BuildContext context) {
+    final SliverMultiBoxAdaptorElement element =
+        context as SliverMultiBoxAdaptorElement;
```

Comment:
do we need both the explicit type and the cast here?

Reply:
nope. removed the `as`.
This code was based on package:flutter code so there are more type checks than we would use.